### PR TITLE
console: migrate AlertBox component to typescript

### PR DESCRIPTION
**Pull request by harishm72**
### Affected components

- [ ] Server
- [x] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

### Related Issues

https://github.com/hasura/graphql-engine/issues/4314#issuecomment-617987200

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

### CODE REVIEW

**Comment by harishm72** [comment]
https://github.com/hasura/graphql-engine/issues/4314#issuecomment-617987200

File: console/src/components/UIKit/atoms/AlertBox/index.tsx [hunk 1]
```diff
@@ -46,7 +51,7 @@ export const AlertBox = props => {
     >
       <Icon type={type} />
       {type && (
-        <Text as="span" pl="md" fontWeight="medium">
+        <Text pl="md" fontWeight="medium">
```
Comment (beerose) [inline console/src/components/UIKit/atoms/AlertBox/index.tsx:54]: Can you bring back `as="span"`? This is a part of the implementation.

File: console/src/components/UIKit/atoms/AlertBox/index.tsx [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-import { theme } from '../../theme';
-import { Icon } from '../Icon';
-import { StyledAlertBox } from './AlertBox';
+import { theme, Theme } from '../../theme';
+import { Icon, IconProps } from '../Icon';
+import { StyledAlertBox, StyledAlertBoxProps } from './AlertBox';
 import { Text } from '../Typography';
 
-const alertBoxWidth = 866;
+const alertBoxWidth = "866";
```
Comment (beerose) [inline console/src/components/UIKit/atoms/AlertBox/index.tsx:8]:
```suggestion
const alertBoxWidth = 866;
```

Comment (harishm72) [reply console/src/components/UIKit/atoms/AlertBox/index.tsx:8]: Sure I will and update the PR.

File: console/src/components/UIKit/atoms/AlertBox/index.tsx (hunk 1 above)
Comment (harishm72) [reply console/src/components/UIKit/atoms/AlertBox/index.tsx:54]: What will be the type of `as` prop,

File: console/src/components/UIKit/atoms/AlertBox/index.tsx [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
+import { theme, Theme } from '../../theme';
+import { Icon, IconProps } from '../Icon';
+import { StyledAlertBox, StyledAlertBoxProps } from './AlertBox';
 import { Text } from '../Typography';
 
 const alertBoxWidth = 866;
 
-export const AlertBox = props => {
+export interface AlertBoxProps extends IconProps, Omit<StyledAlertBoxProps, 'size'> {
+  type: keyof Theme['alertBox'] ;
```
Comment (beerose) [inline console/src/components/UIKit/atoms/AlertBox/index.tsx:11]:
```suggestion
  type: keyof Theme['alertBox'];
```

File: console/src/components/UIKit/atoms/AlertBox/AlertBox.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  FlexboxProps,
+  ColorProps,
+  BorderProps,
+  TypographyProps,
+  LayoutProps,
+  SpaceProps,
+  ShadowProps,
+} from 'styled-system';
+
+import{ BoxProps, Box } from '../Box';
```
Comment (beerose) [inline console/src/components/UIKit/atoms/AlertBox/AlertBox.ts:19]:
```suggestion
import { BoxProps, Box } from '../Box';
```

File: console/src/components/UIKit/atoms/AlertBox/index.tsx [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
+import { Icon, IconProps } from '../Icon';
+import { StyledAlertBox, StyledAlertBoxProps } from './AlertBox';
 import { Text } from '../Typography';
 
 const alertBoxWidth = 866;
 
-export const AlertBox = props => {
+export interface AlertBoxProps extends IconProps, Omit<StyledAlertBoxProps, 'size'> {
+  type: keyof Theme['alertBox'];
+}
```
Comment (beerose) [inline console/src/components/UIKit/atoms/AlertBox/index.tsx:12]:
```suggestion
export interface AlertBoxProps
  extends IconProps,
    Omit<StyledAlertBoxProps, 'size'> {
  type: keyof Theme['alertBox'];
}
```

File: console/src/components/UIKit/atoms/AlertBox/index.tsx [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
 const alertBoxWidth = 866;
 
-export const AlertBox = props => {
+export interface AlertBoxProps
+  extends IconProps,
+    Omit<StyledAlertBoxProps, 'size'> {
+  type: keyof Theme['alertBox'];
+}
+
+
```
Comment (beerose) [inline console/src/components/UIKit/atoms/AlertBox/index.tsx:16]:
```suggestion
```

**Comment by marionschleifer** [comment]
@harishm72 thank you so much for your contribution 🙏 as a thank you, we'd love to send you some swag. If you're interested, please send an email to marion@hasura.io, and we'll take it from there 🙂

**Pull request merged**